src/config.rs

```diff
@@ -229,17 +229,21 @@ macro_rules! make_config {
                     inner.config.clone()
                 };
 
-                /// We map over the string and remove all alphanumeric, _ and - characters.
-                /// This is the fastest way (within micro-seconds) instead of using a regex (which takes mili-seconds)
                 fn _privacy_mask(value: &str) -> String {
-                    value.chars().map(|c|
-                        match c {
-                            c if c.is_alphanumeric() => '*',
-                            '_' => '*',
-                            '-' => '*',
-                            _ => c
+                    let mut masked: String = String::new();
+                    let mut n = 0;
+                    for c in value.chars() {
+                        if c == ',' {
+                            masked += &c.to_string();
                         }
-                    ).collect::<String>()
+                        if n <= 12 && [':', '/'].contains(&c) {
+                            masked += &c.to_string();
+                        } else {
+                            masked += &'*'.to_string();
+                        }
+                        n += 1;
+                    }
+                    masked
```

Comment:
That seems verify inefficient.
I suggest the following change, which is much quicker.

```suggestion
    let mut n = 0;
    value.chars().map(|c| {
        n += 1;
        match c {
            ':' if n <= 11  => c,
            '/' if n <= 13  => c,
            ',' => c,
            _ => '*'
        }
    }
    ).collect::<String>()
```

Reply:
Ah, this is great. Thank you. Yep, I am new to Rust so the Rust way is still - well new.

However, I do have a few comments:

- I never had a `collect` statement in my code, yet in your suggestion it seems that way. how is this possible?
- I was wondering how I could use `map` with an index. now I know. Thanks!
- while your code looks a lot cleaner (and I love it), from a performance perspective the code is not so different. Both use a loop and both use if statements. It is true that I am using one more if, however this and the rest (var assignments) is something the compiler should be able to optimize.

Btw, my code also had a bug, so all of the code was contributed by you and I should not be credited for anything in this PR. I suggest to close this PR and you create a new PR for your code.

Reply:
Thanks for the `godbolt.org` links. This is a great tool. Yep, the memory allocations have more impact than I thought.

> But, you did trigger thinking about it 😄 .

Unfortunately, an idea alone does not make one rich.